crates/holochain_p2p/src/types/actor.rs

```diff
@@ -382,11 +382,13 @@ pub trait HcP2p: 'static + Send + Sync + std::fmt::Debug {
     /// Dump network metrics.
     fn dump_network_metrics(
         &self,
-        dna_hash: Option<DnaHash>,
-    ) -> BoxFut<'_, HolochainP2pResult<String>>;
+        request: Kitsune2NetworkMetricsRequest,
+    ) -> BoxFut<'_, HolochainP2pResult<HashMap<DnaHash, Kitsune2NetworkMetrics>>>;
 
-    /// Dump network stats.
-    fn dump_network_stats(&self) -> BoxFut<'_, HolochainP2pResult<String>>;
+    /// This is untyped because the returned data is backend-specific.
+    ///
+    /// See [Transport::dump_network_stats](kitsune2_api::Transport).
```

Comment:
Reminder to add an doc comment showing the data structure with some example json -- probably makes sense for that to be in the linked kitsune function.

Reply:
Went one better, this is now typed - I hadn't realized that this response doesn't change when you swap out the tx5 backend, only if you replace tx5 with something else. So if somebody replaces the transport implementation that Holochain is using with K2 by patching Holochain then they'd have to fill in some fields about who they're connected to. Yes that means there's a webrtc flag exposed... So it's not the best encapsulation but I think typed is better than not and we're pretty committed to webrtc by default.

Reply:
awesome! looks like the doc comment needs to be updated to reflect that change.

Reply:
Yes, it certainly does. I went through and fixed a bunch of references to JSON - I'll do a self review now and see if I've missed any more

Reply:
I had indeed missed yet more JSON references, I believe I've got them all out now :)